aspnetcore/performance/caching/output.md

```diff
@@ -0,0 +1,155 @@
+
+## Configure one endpoint or page
+
+For minimal API apps, configure an endpoint to do caching by calling `CacheOutput()` or by applying the `[OutputCache]` attribute, as shown in the following examples:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="oneendpoint":::
+
+For apps with controllers, apply the `[OutputCache]` attribute to the action method. For Razor Pages apps, apply the attribute to the Razor page class.
+
+## Configure multiple endpoints or pages
+
+Create *policies* when calling `AddOutputCaching` to specify caching configuration that applies to multiple endpoints. A policy can be selected for specific endpoints, while a base policy provides default caching configuration for a collection of endpoints.
+
+The following highlighted code configures caching for all of the app's endpoints, with expiration time of 10 seconds. If you don't explicitly configure an expiration time, it's one minute.
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies1" highlight="3":::
+
+The following highlighted code creates two policies, each specifying a different expiration time. Selected endpoints can use the 20 second expiration, and others can use the 30 second expiration.
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies1" highlight="4-5":::
+
+You can select a policy for an endpoint when calling the `CacheOutput` method or using the `[OutputCache]` attribute:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="selectpolicy":::
+
+For apps with controllers, apply the `[OutputCache]` attribute to the action method. For Razor Pages apps, apply the attribute to the Razor page class.
+
+## Default output caching policy
+
+By default, output caching follows these rules
+
+* Only HTTP 200 responses are cached.
+* Only HTTP GET or HEAD requests are cached.
+* Responses that set cookies aren't cached.
+* Responses to authenticated requests aren't cached.
+
+The following code applies all of the default caching rules to all of an app's endpoints:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies3":::
+
+The following code removes these defaults while applying caching to all of an app's endpoints:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies4":::
+
+
+You can override these defaults.
+
+## Specify the cache key
+
+By default, every part of the URL is included as the key to a cache entry -- that is, the scheme, host, port, path, and query string. However, you might want to explicitly control the cache key. For example, suppose you have an endpoint that returns a unique response for each unique value of the `culture` query string. Variation in other parts of the URL, such as other query strings, shouldn't result in different cache entries. You can specify such rules in a policy, as shown in the following highlighted code:
```

Comment:
I'm the last person to suggest a punctuation change, but I'm not sure how `--` MT. Maybe ????

```suggestion
By default, every part of the URL is included as the key to a cache entry, that is, the scheme, host, port, path, and query string. However, you might want to explicitly control the cache key. For example, suppose you have an endpoint that returns a unique response for each unique value of the `culture` query string. Variation in other parts of the URL, such as other query strings, shouldn't result in different cache entries. You can specify such rules in a policy, as shown in the following highlighted code:
```